### ISSUE TITLE
UNION with nested ORDER BY timestamp fails with internal error

### ISSUE THREAD

**Issue by luiscleto** [cockroachdb/cockroach#73084]
**Describe the problem**

Some UNION statements seem to fail with an internal error when nested selects rely on order by certain fields (see below for additional details/reproduction):
```
ERROR: internal error: column choice group should have at least one column id
SQLSTATE: XX000
DETAIL: stack trace:
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/props/ordering_choice.go:978: AnyID()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/props/ordering_choice.go:237: ToOrdering()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/ordering/set.go:104: StreamingSetOpOrdering()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:1499: buildSetOp()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/relational.go:333: buildRelational()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:209: build()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/builder.go:171: Build()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:587: runExecBuilder()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:262: makeOptimizerPlan()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1262: makeExecPlan()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:974: dispatchToExecutionEngine()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:712: execStmtInOpenState()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:130: execStmt()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1692: func1()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1694: execCmd()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1616: run()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:660: ServeConn()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:648: func1()
/usr/local/go/src/runtime/asm_amd64.s:1371: goexit()

HINT: You have encountered an unexpected error.

Please check the public issue tracker to check whether this problem is
already tracked. If you cannot find it there, please report the error
with details by creating a new issue.

If you would rather not post publicly, please contact us directly
using the support form.

We appreciate your feedback.

```

**To Reproduce**

For the following table and data:
```
CREATE TABLE IF NOT EXISTS Table1
(
	id INT64 PRIMARY KEY,
	creationDate TIMESTAMP DEFAULT now()
);
insert into Table1 values(50);
insert into Table1 values(51);
```

Running the UNION statement below:
```
(select id from Table1 order by creationDate asc limit 1) union (select id from Table1 order by creationDate desc limit 1);
```

Fails with the error shown above. Ordering by id seems fine although on other tables this has failed with different types of fields as well. I haven't dug very deeply into the issue or what the necessary conditions are to triggering it.

**Expected behavior**
The UNION should have succeeded and returned the two values.

**Environment:**
 - CockroachDB version V21.2.0
 - Server OS: CentOS Linux 8
 - Client app: `cockroach sql`

**Additional context**
Latest release breaks existing queries.

**Comment by rytaft** [comment]
Thank you very much for the report, @luiscleto! Confirmed that I can reproduce the problem using your steps above. We'll work to get this fixed and backported to 21.2.x ASAP.

**Comment by luiscleto** [comment]
Nice. Thanks a lot for the quick turnaround on this!

**Comment by rytaft** [comment]
The fix should most likely be included in 21.2.3
